tests_e2e/orchestrator/lib/agent_test_loader.py

```diff
@@ -96,12 +197,53 @@ def _load_tests(source_file: Path) -> List[Type[AgentTest]]:
         return [v for v in module.__dict__.values() if isinstance(v, type) and issubclass(v, AgentTest) and v != AgentTest]
 
     @staticmethod
-    def _load_file(file: Path):
-        """Helper to load a JSON file"""
+    def _load_images() -> Dict[str, List[VmImageInfo]]:
+        """
+        Loads images.yml into a dictionary where, for each item, the key is an image or image set and the value is a list of VmImageInfos
+        for the corresponding images.
+
+        See the comments in image.yml for a description of the structure of each item.
+        """
+        image_descriptions = AgentTestLoader._load_file(AgentTestLoader._SOURCE_CODE_ROOT/"test_suites"/"images.yml")
+        if "images" not in image_descriptions:
+            raise Exception("images.yml is missing the 'images' item")
+
+        images = {}
+
+        # first load the images as 1-item lists
+        for name, description in image_descriptions["images"].items():
+            i = VmImageInfo()
+            if isinstance(description, str):
+                i.urn = description
+                i.locations = []
+                i.vm_sizes = []
+            else:
+                if "urn" not in description:
+                    raise Exception(f"Image {name} is missing the 'urn' property: {description}")
+                i.urn = description["urn"]
+                i.locations = description["locations"] if "locations" in description else []
+                i.vm_sizes = description["vm_sizes"] if "vm_sizes" in description else []
+            images[name] = [i]
+
+        # now load the image-sets, mapping them to the images that we just computed
+        for image_set_name, image_list in image_descriptions["image-sets"].items():
+            # the same name cannot denote an image and an image-set
+            if image_set_name in images:
```

Comment:
We currently only have one image_set which is 'endorsed', is that correct?

Reply:
for now, yes. the bvts use 'endorsed'. dcr has many more images and image sets, but i don't want to migrate them in bulk. we'll migrate them as we migrate the tests that actually used them.